r-package/policytree/src/tree_search.cpp

```diff
@@ -374,5 +423,66 @@ std::unique_ptr<Node> tree_search(int depth, int split_step, size_t min_node_siz
     v.resize(num_points + 1, 0.0);
   }
 
-  return find_best_split(sorted_sets, depth, split_step, min_node_size, data, sum_array);
+  return find_best_split(sorted_sets, depth, split_step, min_node_size, data, sum_array, 0);
+}
+
+std::unique_ptr<Node> tree_search_hybrid(int max_global_depth, int complete_split_depth, int chop_depth, int repeat_splits, int split_step, size_t min_node_size, const Data* data) {
+  size_t num_rewards = data->num_rewards();
+  size_t num_points = data->num_rows;
+  auto sorted_sets = create_sorted_sets(data);
+
+  std::vector<std::vector<double>> sum_array;
+  sum_array.resize(num_rewards);
+  for (auto& v : sum_array) {
+    // + 1 because this is a cumulative sum of rewards, entry 0 will be zero.
+    v.resize(num_points + 1, 0.0);
+  }
+
+  auto start = new Node(0, 0.0, 0, 0, 0, 0);
+      //TODO(kanodiaayush):uncomment
+  start->complete_sorted_sets = sorted_sets;
+  std::queue<Node*> expansion_queue;
+  expansion_queue.push(start);
+  while(!expansion_queue.empty()) {
+    auto expansion_node = expansion_queue.front();
+    expansion_queue.pop();
+    if(expansion_node->height < 1) {
+	    continue;
+    }
+    if(expansion_node->depth >= max_global_depth) {
+	    continue;
+    }
+      //TODO(kanodiaayush):uncomment
+    //auto expansion_tree = find_best_split(sorted_sets, complete_split_depth, split_step, min_node_size, data, sum_array, expansion_node->depth).release();
+    auto expansion_tree = find_best_split(expansion_node->complete_sorted_sets, complete_split_depth, split_step, min_node_size, data, sum_array, expansion_node->depth).release();
+    std::queue<Node*> bfs_queue;
+    if (expansion_tree->left_child != nullptr) {
+    	bfs_queue.push(expansion_tree->left_child.release());
+    }
+    if (expansion_tree->right_child != nullptr) {
+    	bfs_queue.push(expansion_tree->right_child.release());
+    }
+    int expansion_tree_height = expansion_tree->height;
+    while(!bfs_queue.empty()) {
+    	auto bfs_node = bfs_queue.front();
+      bfs_queue.pop();
+    	if (expansion_tree_height - bfs_node->height == chop_depth) {
+  	  expansion_queue.push(bfs_node);
+	} else if (expansion_tree_height - bfs_node->height < chop_depth) {
+          if (bfs_node->left_child != nullptr) {
+          	bfs_queue.push(bfs_node->left_child.release());
+          }
+          if (bfs_node->right_child != nullptr) {
+          	bfs_queue.push(bfs_node->right_child.release());
+          }
+	}
+    }
+    expansion_node = expansion_tree;
+    //expansion_node = std::move(expansion_tree);
+    //if (expansion_node->depth == 0) {
+      //start = std::move(expansion_node);
+    //}
+  }
+
+  return std::unique_ptr<Node> (start);
```

Comment:
just tacking on a comment to @kanodiaayush that this line is iffy doing everything with unique ptrs since `start` ceases to exist once it's pushes onto the queue